File: Java/Solutions/3.2. Simple Conditions - Exam Problems/Harvest.java
import java.util.Scanner;

public class Harvest {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double vineyardArea = Double.parseDouble(scanner.nextLine());
        double grapePerSquare = Double.parseDouble(scanner.nextLine());
        double neededLiters = Double.parseDouble(scanner.nextLine());
        double workers = Double.parseDouble(scanner.nextLine());

        double harvestPerVine = (vineyardArea * grapePerSquare) * 0.4;
        double vine = harvestPerVine / 2.5;

        if (vine >= neededLiters) {
            double vineLeft = vine - neededLiters;
            System.out.printf("Good harvest this year! Total wine: %.0f liters.%n",
                    Math.floor(vine));
            System.out.printf("%.0f liters left -> %.0f liters per person.",
                    Math.ceil(vineLeft), Math.ceil(vineLeft / workers));
        } else {
            System.out.printf("It will be a tough winter! More %.0f liters wine needed.",
                    Math.floor(neededLiters - vine));
        }
    }
}

File: JavaScript/Solutions/03.Simple-conditions-exam-problems/Harvest.js
function harvest(
    [vineyardArea, grapePerSquare, neededLiters, workers]) {

    let harvestPerVine = (vineyardArea * grapePerSquare) * 0.4;
    let vine = harvestPerVine / 2.5;

    if (vine >= neededLiters) {
        let vineLeft = vine - neededLiters;
        console.log(`Good harvest this year! Total wine: ${Math.floor(vine)} liters.`);
        console.log(`${Math.ceil(vineLeft)} liters left -> ${Math.ceil((vineLeft / workers))} liters per person.`);
    }
    else {
        console.log(`It will be a tough winter! More ${Math.floor(neededLiters - vine)} liters wine needed.`);
    }
}
